src/main/java/com/checkmarx/ast/ScanResult/ScanResult.java

```diff
@@ -0,0 +1,50 @@
+package com.checkmarx.ast.ScanResult;
+
+import com.checkmarx.ast.wrapper.CxBaseObject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import lombok.Value;
+
+import java.util.List;
+import java.util.Map;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+@JsonDeserialize()
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ScanResult extends CxBaseObject {
```

Comment:
why are you extending CxBaseObject if you don't use any of his properties?

Reply:
It uses the method "parse" that's defined there..
if we really want to get rid of this inheritance we can take "parse" and the method it uses - "isValidJson" to JSON utils, because these two method don't seem related to CxBaseObject logically.
What do you think?

Reply:
I think it better idea then have empty base clas fields